app/src/main/java/org/desarrolladorslp/teckersapp/ui/programs/ProgramBatchFragment.kt

```diff
@@ -0,0 +1,174 @@
+package org.desarrolladorslp.teckersapp.ui.programs
+
+import android.content.Context
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.*
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.view.isVisible
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProviders
+import com.google.android.material.floatingactionbutton.FloatingActionButton
+import kotlinx.android.synthetic.main.fragment_programs_batches.*
+import org.desarrolladorslp.teckersapp.R
+import org.desarrolladorslp.teckersapp.model.Batch
+import org.desarrolladorslp.teckersapp.model.Program
+import org.desarrolladorslp.teckersapp.ui.batches.BatchViewModel
+import org.desarrolladorslp.teckersapp.ui.batches.BatchesFragment
+import org.desarrolladorslp.teckersapp.ui.teckers.TeckersFragment
+import java.lang.IllegalStateException
+
+class ProgramBatchFragment: Fragment() {
+
+    lateinit var programSpinner: Spinner
+    lateinit var programsViewModel:ProgramViewModel
+    lateinit var batchSpinner: Spinner
+    lateinit var programTextView:TextView
+    lateinit var batchTextView: TextView
+    lateinit var batchesViewModel: BatchViewModel
+    lateinit var approvalButton : FloatingActionButton
+    var selectedProgram: Program? = null
+    var selectedBatch: Batch? = null
+    lateinit var listener :BatchSelectedListener
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        programsViewModel =
+            ViewModelProviders.of(parentFragment!!).get(ProgramViewModel::class.java)
+        batchesViewModel =
+            ViewModelProviders.of(parentFragment!!).get(BatchViewModel::class.java)
+    }
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+
+        val  root= inflater.inflate(R.layout.fragment_programs_batches, container, false)
+        programSpinner = root.findViewById(R.id.spinner_programs) as Spinner
+        batchSpinner = root.findViewById(R.id.spinner_batches) as Spinner
+        programTextView = root.findViewById(R.id.text_programs) as TextView
+        batchTextView = root.findViewById(R.id.text_batches) as TextView
+        approvalButton  = root.findViewById(R.id.btn_approval)as FloatingActionButton
```

Comment:
all of these could be referenced by their id with kotlin synthetic properties.. however, if you do this, make sure to move this logic to `onViewCreated`